src/chain.h

```diff
@@ -414,16 +414,9 @@ class CDiskBlockIndex : public CBlockIndex
         return block.GetHash();
     }
 
-
-    std::string ToString() const
```

Comment:
I'm somewhat ambivalent about removing this. The superclass has it so it can be considered part of the API, and it can be useful to show the contents of the structure when debugging/troubleshooting.

Reply:
Should we rename it instead?

(Noting that the safe and correct way would be to use dynamic dispatch, like in https://github.com/jonatack/bitcoin/commits/CBlockIndex-CDiskBlockIndex-class-design, if the base and derived classes aren't critical hotspots.)

Reply:
I agree it's not worth it to set that up just for a debugging function.

Reply:
Right.

Reply:
Resolving as the method is unused and can have inconsistent behavior as described in the OP, but please let me know if a change should be made.